source/request.test.js

```diff
@@ -0,0 +1,212 @@
+const Request = require('./request');
+
+test('preprocesses headers', () => {
+  let details = {
+    responseHeaders: [
+      {
+        name: 'server',
+        value: 'cloudflare'
+      }
+    ]
+  };
+
+  let r = new Request(details);
+
+  expect(r.headers).toMatchObject({
+    SERVER: 'cloudflare'
+  });
+});
+
+describe('Railgun', () => {
+  test('processes Railgun header v1', () => {
+    let details = {
+      responseHeaders: [
+        {
+          name: 'cf-railgun',
+          value: 'normal 96 foobar 9001'
+        }
+      ]
+    };
+
+    let r = new Request(details);
+
+    expect(r.railgunMetaData).toMatchObject({
+      normal: true,
+      id: 'normal',
+      version: '9001',
+      flags: 0x60,
```

Comment:
How did this turn into a hex value? I only see `parseInt` with decimal (`10`) as second argument in the [Railgun header processing logic](https://github.com/cloudflare/claire/blob/master/source/request.js#L34-L108).

Reply:
It's parsed and compared one and the same. Having it hex here is easier for me, since it's an AND of the individual flags.